folding/validators/protein.py

```diff
@@ -526,34 +569,82 @@ def process_md_output(self, md_output: Dict, hotkey: str) -> bool:
             files=md_output,
             output_directory=self.miner_data_directory,
         )
+        try:
+            simulation = self.recreate_simulation(
+                seed, state, f"{self.miner_data_directory}/{md_outputs_exts['cpt']}"
+            )
+        except Exception as e:
+            bt.logging.error(f"Failed to recreate simulation: {e}")
+            return False
 
-        last_miner_simulation_step_time = get_last_step_time(
-            os.path.join(self.miner_data_directory, md_outputs_exts["log"])
+        cpt_step = simulation.currentStep
+        log_file = pd.read_csv(f"{self.miner_data_directory}/{md_outputs_exts['log']}")
+        log_step = log_file['#"Step"'].iloc[-1]
+
+        ## Make sure that we are enough steps ahead in the log file compared to the checkpoint file.
+        if log_step - cpt_step < 5000:
+            bt.logging.error("Miner did not run enough steps since last checkpoint")
+            return False
+
+        return True
+
+    def recreate_simulation(
```

Comment:
I also don't think I understand why this is called "recreate" and not "create". Is this a second step?

Reply:
It is recreating the simulation based on the miners checkpoint